esx_service/utils/auth_api.py

```diff
@@ -22,42 +22,68 @@
 import error_code
 import log_config
 import logging
+from error_code import ErrorCode
+from error_code import ErrorInfo
+
+def generate_error_info_for_internal_error(secondary_err_msg):
```

Comment:
It's not the same thing. If you look at these 3 new utility methods, they are doing the same thing:
1. Get the error_code
2. Call error_code.error_codes_to_messages() API to get a user-friendly error message
3. Contruct an ErrorInfo object and return
My suggestion is to add a new API in error_code to do these common logic so that we don't need to introduce these 3 methods which contains duplicate logic.